### PR TITLE
Add spotlessCheck job to the security CI workflow

### LINKED ISSUE #9023: Verify Spotless in CI
### Problem

Sometimes spotless doesn't run on a PR before it's merged to main despite it being configured to run automatically as part of a git hook. This can cause other's PRs that merge from main afterwards to automatically run spotless against files they didn't change.

### Solution

Add a check in security CI that runs `spotlessCheck` and fails the check if any files need to be changed by Spotless

### Alternatives

_No response_

### PR DESCRIPTION
**Pull request by sdimitrov9**
**Description**:
This PR adds spotlessCheck job to the security CI workflow.

**Related issue(s)**:

Fixes #9023 

**Checklist**

- [ ] Documented (Code comments, README, etc.)
- [ ] Tested (unit, integration, etc.)

### CODE REVIEW

**Review by edwin-greene — APPROVED** [review]
Looks good to me

File: .github/workflows/security.yml [hunk 1]
```diff
@@ -241,3 +241,41 @@ jobs:
             html-to-markdown ${report}.html -o build/reports && \
             cat ${report}.html.md >> $GITHUB_STEP_SUMMARY
           fi
+
+  spotless-check:
+    if: github.event_name == 'push' || (github.event.pull_request.head.repo.full_name == github.repository && github.actor != 'dependabot[bot]')
```
Comment (steven-sheehy) [inline .github/workflows/security.yml:246]: Remove this as this doesn't use any secret and we want it  to always run

File: .github/workflows/security.yml [hunk 2]
```diff
@@ -241,3 +241,41 @@ jobs:
...
+    name: Spotless Code Format Check
+    runs-on: mirror-node-linux-large
```
Comment (steven-sheehy) [inline .github/workflows/security.yml:248]: This should probably run on mirror-node-linux-medium

File: .github/workflows/security.yml [hunk 3]
```diff
@@ -241,3 +241,41 @@ jobs:
...
+    steps:
+      - name: Harden Runner
+        uses: step-security/harden-runner@0080882f6c36860b6ba35c610c98ce87d4e2f26f # v2.10.2
+        with:
+          egress-policy: audit
+
+      - name: Checkout Code
+        uses: actions/checkout@11bd71901bbe5b1630ceea73d27597364c9af683 # v4.2.2
+        with:
+          fetch-depth: 0 # Shallow clones should be disabled for a better relevancy of analysis
```
Comment (steven-sheehy) [inline .github/workflows/security.yml:258]: Use shallow clones to speed it up

File: .github/workflows/security.yml [hunk 4]
```diff
@@ -241,3 +241,41 @@ jobs:
...
+
+      - name: Setup Node
+        uses: actions/setup-node@39370e3970a6d050c480ffad4ff0ed4d3fdee5af # v4.1.0
+        with:
+          node-version: 21
```
Comment (steven-sheehy) [inline .github/workflows/security.yml:263]: Remove as it shouldn't need node

**Review by steven-sheehy — CHANGES_REQUESTED** [review]
Need to merge from main and verify spotless check now passes

File: .github/workflows/security.yml (hunk 3 above)
Comment (sdimitrov9) [reply .github/workflows/security.yml]: Done, fetch-depth is now set to 10.

**Comment by sdimitrov9** [comment]
> Need to merge from main and verify spotless check now passes

Done.

File: .github/workflows/security.yml [hunk 5]
```diff
@@ -241,3 +241,35 @@ jobs:
...
+        with:
+          distribution: temurin
+          java-version: 21
+
+      - name: Setup Gradle
+        uses: gradle/actions/setup-gradle@cc4fc85e6b35bafd578d5ffbc76a5518407e1af0 # v4
+
+      - name: Execute Gradle
+        env:
+          GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
```
Comment (steven-sheehy) [inline .github/workflows/security.yml:270]: Remove this secret

**Review by steven-sheehy — CHANGES_REQUESTED** [review]

File: .github/workflows/security.yml (hunk 5 above)
Comment (sdimitrov9) [reply .github/workflows/security.yml]: Done.

**Review by steven-sheehy — APPROVED** [review]

**Pull request closed without merging**